wikimedia/wikipedia-ios
# Add Edit Attempt Schema Funnel

**Phabricator:** https://phabricator.wikimedia.org/T313561

### Notes
* This PR adds logging for edit actions on the edit attempt schema - can be found [here](https://github.com/wikimedia/schemas-event-secondary/blob/master/jsonschema/analytics/legacy/editattemptstep/current.yaml) 
* Our implementation was based on Android -  [for reference](https://github.com/wikimedia/apps-android-wikipedia/blob/main/app/src/main/java/org/wikipedia/analytics/eventplatform/EditAttemptStepEvent.kt) 
* Also to maintain some parity with android, we're not sending `user_editcount`
* Some required fields, like MediaWiki version, are not available for us, so we're sending empty literals in this case

### Test Steps
1. Go to any article, and click the edit button. 
2. Go all the way to confirming your edit
3. Start another edit attempt, but this time cancel it 
4. Check if the events are being sent
